### DISCUSSION TITLE
How do you handle tests folders ?

### DISCUSSION THREAD

**Discussion by mehdigmira** [microsoft/pyright#4924] [Q&A]
Hello,

Let's say I have the following project:

```
pyproject.toml
src/
tests/
```

I'd like to enable pyright type checking on src/ folder of course, but not on tests/ since this would make writing tests painful.
I'd still like to benefit from pylance features (autocomplete, autoimport, lookup symbol) for tests though. 
It seems to me this is a very usual usecase, but i can't find a straightforward way to do this:
- if i do not add tests/ to include in pyproject.toml, or if i exclude it, then lookup symbol feature does not work
- i i include it but mark it as ignored, then auto-import doesn't work

what's the best way to do this ?

**Comment by erictraut** [comment]
If you want to exclude this directory from being analyzed when running the CLI version of pyright, you can add an `exclude` section to your pyright config file. Note that pyright and pylance will still analyze the files in this directory and report any problems if you open them in the editor. 

If you want to completely suppress all of the errors for the files in this directory — both in the CLI version of pyright and in pylance, you can add an `ignore` section to your pyright config file.

Adding a file or directory to the `ignore` section shouldn't affect language server features like hover text, completion suggestions, etc. It simply suppresses all diagnostics from those files.

**Reply by mehdigmira** [reply]
Thanks for your answer. But I'm not sure this entirely fixes the issue for me:
- If i add an exclude section, as I said, the symbols of the excluded modules are no longer available in the "symbol table" of VScode, which makes code navigation a bit tedious (basically having to cmd+shift+F to find symbols)
- If i add it to ignore completion suggestions do work, But not quick fix, which I often use to autoimport unknown symbols in lines of code that I have copied paste for example

**Reply by MrGreenTea** [reply]
That's what I do, but it is an unsatisfying solution to me.  I then use `ruff-lsp` to help me with some of the more critical diagnostics like undefined names and unused imports.

I'd love it if I could specify specific settings for my `tests` folder like enabling and disabling specific diagnostics for example.
